resources/puppetlabs/lein-ezbake/template/global/ext/fpm.rb

```diff
@@ -181,25 +182,28 @@
     options.systemd_el = 1
   elsif options.operating_system == :el && options.os_version >= 7 # systemd el
     if ! options.is_pe
+      # https://bugzilla.redhat.com/show_bug.cgi?id=2224427
       fpm_opts << "--depends tzdata-java"
-      options.java =
-        case options.platform_version
-        when 8
-          # rpm on Redhat 7 may not support OR dependencies
-          if options.os_version == 7
-            'java-11-openjdk-headless'
-          elsif options.os_version == 8
-            '(java-17-openjdk-headless or java-11-openjdk-headless)'
-          elsif options.os_version == 9
-            'java-17-openjdk-headless'
-          else
-            fail "Unrecognized el os version #{options.os_version}"
-          end
-        when 6..7
-          'java-1.8.0-openjdk-headless'
+      case options.platform_version
+      when 8
+        # rpm on Redhat 7 may not support OR dependencies
+        if options.os_version == 7
+          options.java = 'jre-11-headless'
+          options.java_bin = '/usr/lib/jvm/jre-11/bin/java'
+        elsif options.os_version == 8
+          options.java = '(jre-17-headless or jre-11-headless)'
+          # TODO: which bin to use? /usr/bin/java may be anything
+        elsif options.os_version == 9
+          options.java = 'jre-17-headless'
```

Comment:
You can question why EL 9 always uses Java 17 while EL 8 uses 17 or 11. Both EL 8 and EL 9 provide both versions.